launcher/ui/pages/global/JavaPage.cpp

```diff
@@ -62,141 +62,35 @@
 JavaPage::JavaPage(QWidget* parent) : QWidget(parent), ui(new Ui::JavaPage)
 {
     ui->setupUi(this);
+    
+    m_javaSettings = new JavaSettingsWidget(nullptr, this);
+    ui->generalScrollArea->setWidget(m_javaSettings);
```

Comment:
Wouldn't it be better in the '.ui' file?
If not at least deleteLater in destructor

Reply:
It should be able to add a constructor with only the parent widget that calls this constructor under the hood.
Then with that constructor move it to the UI file